### PR TITLE
Vlad task3

### PR DESCRIPTION
**Pull request by vlad-iancu**

### CODE REVIEW

File: vlad_task3/SConscript [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+""" main_libs = [lib]
+if test:
+    main_libs.append("gcov")
+ """
+#common_lib = env.Library("common", ["src/Common.cpp"])
+env.Program("practice", ["practice.cpp"])
+proxy_base_lib = env.Library("proxy_base",["src/ProxyBase.cpp"])
+#dynamic_array_lib = env.Library("dynamic_array", ["src/DynamicArray.cpp"])
+env.Program("main_server", ["main_server.cpp", "src/DynamicArray.cpp", "src/ServerProxyObject.cpp"], LIBS=[proxy_base_lib, "$LIBS"])
+#client_lib = env.Library("client", ["src/ClientProxyObject.cpp"])
```
Comment (maxlig79) [inline vlad_task3/SConscript]: Please remove unneeded lines.

File: vlad_task3/practice.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    ShmString *str = managed_shm.find_or_construct<ShmString>("String")("New ", managed_shm.get_segment_manager());
+    const char *cstr = (*str).c_str();
+    str->insert(3, " string");
+    std::cout << *str << std::endl;
+    std::cout << cstr << std::endl;
+
+    return 0;
+
+}
+/* #include <boost/interprocess/managed_shared_memory.hpp>
```
Comment (maxlig79) [inline vlad_task3/practice.cpp]: Do we need this file?

File: vlad_task3/src/ServerProxyObject.cpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            }
+            break;
+            case ADD:
+            {
+                {
+                    boost::interprocess::scoped_lock<boost::interprocess::named_mutex> lk(*mutex);
+                    arr->addEntry(getValue());
+                    setStatus(SUCCESS);
+                }
+                cond->notify_all();
```
Comment (maxlig79) [inline vlad_task3/src/ServerProxyObject.cpp]: The condition notify occur in every case, maybe you can write it after the switch that should include also a default for wrong commands.

File: vlad_task3/src/ServerProxyObject.cpp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+using namespace common;
+
+namespace server
+{
+    ServerProxyObject::ServerProxyObject(DynamicArray *arr) : ProxyBase(boost::interprocess::create_only), arr(arr)
+    {
+    }
+
+    void ServerProxyObject::run()
+    {
```
Comment (maxlig79) [inline vlad_task3/src/ServerProxyObject.cpp:15]: Why the server has a run method while the client handle this in the main?

**Review by maxlig79 — CHANGES_REQUESTED** [review]

**Review by maxlig79 — APPROVED** [review]

**Review by ArdeleanuStefan — APPROVED** [review]

**Pull request merged**